spec/core/UtilSpec.js

```diff
@@ -31,6 +31,36 @@ describe("jasmineUnderTest.util", function() {
     });
   });
 
+  describe("isPromise", function () {
+
+    it("should return true when passed a native promise", function () {
+      var p = new Promise(function (res, rej) {});
+      expect(jasmineUnderTest.isPromise(p)).toBe(true);
+    });
+
+    it("should return false for promise like objects", function () {
+      var promiseLikeObject = function () {this.then = function () {};};
+      var p = new promiseLikeObject();
+      expect(jasmineUnderTest.isPromise(p)).toBe(false);
+    });
+
+  });
+
+  describe("isPromiseLike", function () {
```

Comment:
We should also have some negative tests for `isPromiseLike` as well. Something like checking for a `then` that isn't actually a function, and that `null` and `undefined` continue to work.

Reply:
Interestingly I added tests for isPromise as well and found a bug. Currently the functionality is

```javascript
    return typeof jasmineGlobal.Promise !== 'undefined' && obj && obj.constructor === jasmineGlobal.Promise;
```

this totally looks like it should work, but if obj is null or undefined, that value (null or undefined) will be returned.

Not super sure why this happens, but if you evaluate

```javascript
1 === 1 && null && 1 === 1
```

you will get null.

I'm changing this method to 

```javascript
return typeof jasmineGlobal.Promise !== 'undefined' && !!obj && obj.constructor === jasmineGlobal.Promise;
```

and now it works as expected.